src/main/scala/ch/epfl/bluebrain/nexus/kg/resolve/CompositeResolution.scala

```diff
@@ -0,0 +1,51 @@
+package ch.epfl.bluebrain.nexus.kg.resolve
+import cats.Monad
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import ch.epfl.bluebrain.nexus.kg.resources.{Ref, Resource}
+
+/**
+  * Implementation that uses underlying resolutions in defined order to resolve resources.
+  * @param  resolutions underlying resolutions
+  * @tparam F the resolution effect type
+  */
+class CompositeResolution[F[_]](resolutions: List[Resolution[F]])(implicit F: Monad[F]) extends Resolution[F] {
```

Comment:
We can use the context bound notation in this class for clarity
`F[_]: Monad`

Reply:
He used the implicit notation because he's using `F.pure` in the code. I think in this case it is fine